### ISSUE TITLE
release prep

### ISSUE THREAD

**Issue by brentp** [arq5x/gemini#912]
It's time to make a new release this issue will track what needs to be done. Please add any essential (and only essential) updates as comments to this issue:

+ [ ] update clinvar
+ [ ] update dbsnp
+ [ ] update gnomad
+ [ ] remove dgidb stuff as it's no longer working or supported.

**Comment by jxchong** [comment]
presumably issue #903 is included

**Comment by jxchong** [comment]
When you update gnomad, will you include the subcohorts (i.e. frequencies from the controls-only and non-neuro cohorts)?

**Comment by brentp** [comment]
I updated the list to include those items. thanks for the reminders

**Comment by pfpjs** [comment]
Taking _de novos_ into account in the _comp_het_ inheritance model would be nice!

**Comment by arq5x** [comment]
I think it would be great to add support for de novos when reporting candidate CHs. The challenge is that one cannot be confident that it causes a true CH without read-backed phasing in the common case of trios. As such, this would have to either be part of the relaxed priority CH searches or a specific mode.

**Comment by pfpjs** [comment]
Yeah, the variant could be on the same allele as the inherited one and thus not be a real comp_het, but perhaps a new lenient mode, or a flag for these specific cases would help. Also, this applies for diploid organisms (human specifically, in my case), for other ploidies, things could get ugly fast.

**Comment by jxchong** [comment]
Agree it'd be nice. I've used this hack to get at the results:

Run `comp_hets --max-priority 2` which will give you all potential CH pairs including those that are de_novo + 2nd allele. This yields a lot of false positives, but you can winnow down the results by cross-referencing the gene list with the results of `de_novo` and looking for overlapping genes.

**Comment by pfpjs** [comment]
@jxchong, thank you for that hint!

**Comment by brentp** [comment]
I'll see what I can do about comp-het + DN. Though it can't be phased, it could be a special category.

**Comment by bgruening** [comment]
@brentp https://github.com/arq5x/gemini/issues/911 and https://github.com/arq5x/gemini/issues/910 are two regressions from the 0.20 version.

**Comment by brentp** [comment]
@wm75 can you make a PR or let me know the URL? I'm not familiar with dgidb.

**Comment by wm75** [comment]
Sure, just need to get my laptop to look it up.

**Comment by wm75** [comment]
Ok, so you'd use:

`dgidb_url = 'http://www.dgidb.org/api/v2/interactions.json?genes='`

to re-enable things, at least, that's the minimal change.

What I'd prefer though would be if URLs (the dgidb, the install-data url, others?) became part of the config file. This would give, e.g., Galaxy Admins a single place to locate these URLs and update them if they break at some point. I realize that is a slightly bigger change, but if you have a little bit of time before the release, I think it would help a lot with long-term support of any gemini install.

**Comment by wm75** [comment]
@bgruening we've been talking about this before. The above line is the patch to https://github.com/arq5x/gemini/blob/8a3e7571b15cec31f701161f6efe38bc624be028/gemini/dgidb.py#L28
that you'd have to apply to gemini on usegalaxy.eu to make the *actionable mutations* and the --dgidb option of the *query* tool work again.

**Comment by brentp** [comment]
great. thank you!
so the dgidb stuff is getting used? If so, I can keep it as long as it's this simple URL change to fix.

**Comment by wm75** [comment]
@bgruening it's working :smile:

@brentp yeah, we have an ongoing project with clinicians, for whom being able to get at gene-drug interactions is a major selling point. So I'd rather like to see more such functionality than less.

**Comment by brentp** [comment]
> When you update gnomad, will you include the subcohorts (i.e. frequencies from the controls-only and non-neuro cohorts)?

@jxchong are these the only additions from gnomad? I'll put a list of what I plan to add for final verification, but an initial set would be helpful.

**Comment by pfpjs** [comment]
Any possibility of updating CADD to the new v1.4?

**Comment by brentp** [comment]
I have added the CADD update to the list.

**Comment by brentp** [comment]
All, I have implemented the comp-het + denovo (CHDN) in a specific way and I'm looking for problems I might not have foreseen. Currently, the highest priority is 1. I have made a "good" CHDN candidate have a priority of 1.5 so one can filter with that (I will change the gemini parameter to accept floats).

A good candidate for kid, mom, dad would be:
```
# het inherited from mom:
A/T, A/T, A/A
# DN
C/T, C/C, C/C
```

A somewhat arbitrary requirement that I have added is that the DN HET can not occur in any unaffected sample in the family. So, if we add an unaffected sib to the above trio so it's
kid, mom, dad, sib and the sib also has the de novo:
```
```
# het inherited from mom only in proband:
A/T, A/T, A/A, A/A
# DN in both kids
C/T, C/C, C/C, C/T 
```
this is not reported as a candidate since an unaffected (sib) shares the DN. This could miss cases where it was a germ-line mosaic, but I think it will also remove a lot of false-positives.

Any thoughts on this approach? I want to avoid extra flags and arguments and just provide a 95% solution here.

**Comment by oleraj** [comment]
Another possible addition is gnomad exomes update to 2.1: 
http://gnomad.broadinstitute.org/downloads

**Comment by brentp** [comment]
@oleraj that's part of the gnomad item above. I'm testing that now. The file is much larger so I'm trying to cull it a bit.

**Comment by oleraj** [comment]
Will this be a separate genetic model, or will it be part of the existing comp_hets?  The approach seems fine to me for now (others should weigh in too).

**Comment by brentp** [comment]
it will be part of comp-het and not require any change (other than bumping the max-priority to > 1.5

**Comment by brentp** [comment]
@jxchong would you have a look at: https://github.com/arq5x/gemini/commit/513f53fe2a2e07c25af6e7dd27db74ac25388339 ? and verify it meets your needs for the non-neuro and controls AFs? I just took those 2 fields from the VCF and add them to the database.

**Comment by jxchong** [comment]
@brentp commit 513f53f looks good. Do you think others might have a use for the non-cancer control set too? Perhaps yes, especially when analyzing mosaic variants? @oleraj

**Comment by brentp** [comment]
Is anyone against removing ESP? Given it's size relative to newer resources, it would drop a few columns...

**Comment by jxchong** [comment]
Hm, we'd like to see ESP still be included in max_aaf_all (not all ESP samples can be incorporated into gnomAD and if something is relatively common in ESP because of those samples, that's useful info) but don't need to actually output the database-specific frequency.

**Comment by oleraj** [comment]
I wasn't aware of the cancer samples in gnomad.  Just read about this in a previous faq [here](http://gnomad.broadinstitute.org/faq).  Yes, probably would be good to include the non-cancer control set.

**Comment by brentp** [comment]
ok. I have added that. I am building (and re-building) dbsnp and CADD. 
Once those are done, my planned development is done. I will not update the gene_detailed and gene_summary stuff.
If there are any other high-priority issues, please let me know in the next couple of days.

**Comment by pfpjs** [comment]
I may be able to tackle the gene_summary and gene_detailed table updates, but I will only be able to do it during next weekend.

**Comment by brentp** [comment]
I have updated the inheritance module so anyone interested can test the comp-het + de novo and the transmitted de novo stuff by doing: `pip install -U inheritance` to get v0.1.5 before we make the gemini release.

**Comment by oleraj** [comment]
Some issues from my wish list for potential inclusion:

As you're re-building CADD, is there any way you could include the CADD indels too, preferably merged with the SNPs?  #833 

Related to dbNSFP, would you be able to address #886 ? Basically, I think we can either modify entries in dbNSFP to convert `=` to something else while re-building the database for gemini, or just remove this line from geneimpacts/effect.py 
```
assert not "=" in effect_string
```

We ran into a few errors in running tests last time we tried to install from master, documented in #901, in case there are any quick fixes you might notice. Some of them seem pretty trivial, e.g., gemini/test/unicode.vcf.gz was missing -- probably inadvertently deleted from the repository at some point. 

We also had a discussion in #663 around prioritizing canonical transcripts, which seems to be implemented in geneimpacts>=0.3.4.  If there is a simple way to incorporate the latest version of geneimpacts and maybe add a --prioritize_canonical flag to gemini load, that would be helpful.

**Comment by brentp** [comment]
@oleraj I have responded on a few of those issues and #901 is fixed.
I will incorporate the latest version of geneimpacts when making the release. For others, I'd prefer those that need that functionality to contribute as there are relatively simple work-arounds for those.

**Comment by oleraj** [comment]
Great, thanks @brentp.

**Comment by jxchong** [comment]
Oh! What about including CCRs as an annotation? :)

**Comment by arq5x** [comment]
Agree; CCRs are in the larger plan, but will leave it to Brent if it will make it into this release.

**Comment by pfpjs** [comment]
I've managed to fix and update the generation of `detailed_gene_table` and `summary_gene_table`, can anyone else take a second look? Thanks!

**Comment by Phillip-a-richmond** [comment]
Late to the party but I look forward to the new release!

Will some aspects of the update (non database-related) also remain functional with the VCFAnno + VCF2DB pipeline? Specifically thinking about gene_summary and gene_detailed, and if there's a way to leverage these improvements within my current BCFTools (soft-filter), SNPeff, VCFAnno, VCF2DB pipeline.

Or do you now suggest switching back to the gemini load + gemini annotate to get extra features (e.g. in-house database). 

Thanks!
Phil

**Comment by brentp** [comment]
@Phillip-a-richmond you should continue using the new loader. this will be the last major release using the old loader. If there are features missing when using the new loader, then PR's would be great.

**Comment by brentp** [comment]
dgidb seems to be broken again. perhaps it is an intermittent server error but I don't plan to fix it. I will rely on those that use dgidb to keep that functioning.
